src/auth/auth.controller.ts

```diff
@@ -0,0 +1,135 @@
+
+    @Post('login')
+    @ApiCreatedResponse({
+        description: 'Login via any supported crypto wallet',
+    })
+    @ApiUnauthorizedResponse({
+        description: 'Invalid wallet signature or outdated request',
+    })
+    @ApiBody({
+        required: true,
+        type: LoginRequestDto,
+    })
+    async login(@Body() request: LoginRequest, @Res() res: Response) {
+        if (!this.verifyLoginService.verifySignature(request)) {
+            this.authService.clearJwtCookies(res);
+
+            throw new I18nHttpException(
+                new UnauthorizedException(
+                    'Unauthorized',
+                    'Invalid wallet signature',
+                ),
+                'auth.error.signature',
+            );
+        }
+
+        if (!this.verifyLoginService.verifyTimestamp(request)) {
+            this.authService.clearJwtCookies(res);
+
+            throw new I18nHttpException(
+                new UnauthorizedException(
+                    'Unauthorized',
+                    'The signed request timed out',
+                ),
+                'auth.error.timestamp',
+            );
+        }
+
+        const user: AuthenticatedUser = { wallet: request.wallet };
+
+        this.authService.createAccessToken(user, res);
+        const refreshToken = this.authService.createRefreshToken(user, res);
+
+        await this.sessionRepository.create(<Session>{
+            createdAt: refreshToken.payload.createdAt,
+            walletAddress: refreshToken.payload.wallet,
+            hashedToken: await hash(refreshToken.jwt, 10),
+        });
+
+        res.status(HttpStatus.CREATED).send();
+    }
+
+    @UseGuards(JwtRefreshGuard)
+    @Post('refresh')
+    @ApiOkResponse({
+        description: 'Renews jwt access- and refresh-token',
+    })
+    @ApiUnauthorizedResponse({
+        description: 'Invalid or outdated refresh token',
+    })
+    async refresh(@Req() req: Request, @Res() res: Response) {
+        const user = req.user as RefreshJwtPayload;
+
+        const session = await this.sessionRepository.getByWalletAddressAndCreatedAt(
+            user.wallet,
+            user.createdAt,
+        );
+
+        // Reset timeout timer
+        session.updatedAt = new Date();
+        await this.sessionRepository.update(session);
+
+        this.authService.createAccessToken(user, res);
+        this.authService.renewRefreshToken(session, res);
+
+        res.status(HttpStatus.OK).send();
+    }
+
+    @Post('logout')
+    @UseGuards(JwtRefreshGuard)
+    @ApiOkResponse({
+        description: 'Clears cookies and invalidates jwt refresh token',
+    })
+    async logout(@Req() req: Request, @Res() res: Response) {
+        const user = req.user as RefreshJwtPayload;
+
+        await this.sessionRepository.deleteByWalletAddressAndCreatedAt(
+            user.wallet,
+            user.createdAt,
+        );
+
+        this.authService.clearJwtCookies(res);
+
+        res.status(HttpStatus.OK).send();
+    }
+}
```

Comment:
I understand your point here. So what about the future development of the project? Once we have other features that also require authentication but are not directly related to account management, I think it would be okay to keep them in a separate folder.